okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -121,8 +123,8 @@ public static RealConnection testConnection(
     return result;
   }
 
-  public void connect(
-      int connectTimeout, int readTimeout, int writeTimeout, boolean connectionRetryEnabled) {
+  public void connect(int connectTimeout, int readTimeout, int writeTimeout,
+      boolean connectionRetryEnabled, Call call, EventListener eventListener) {
```

Comment:
Initially I put these as final fields on RealConnection. A few tests failed and it made me realize it's probably not a great idea for a connection to have a strong reference to the call or the event listener.